sections/06_discussion.md

```diff
@@ -49,6 +49,89 @@ with only a couple GPUs.*
 *Some of this is also outlined in the Categorize section.  We can decide where
 it best fits.*
 
+Efficiently scaling deep learning is challenging, and there is a high
+computational cost (e.g., time, memory, energy) associated with training neural
+networks and using them for classification. For these reasons, neural networks
+have only recently found widespread use [@tag:Schmidhuber2014_dnn_overview].
+For biologists, such problems are further complicated by the immense size of
+most biological datasets.
+
+*TODO: Perhaps a visual illustrating the changes over time in the cost of
+storage
+(i.e. hard drive space), genome sequencing, and processing power. Additionally,
+plotting the accuracy, depth, and training time of imagenet winners could be
+illustrative. See [@tag:Stein2010_cloud] for example of calculating storage
+cost,
+and [@tag:NIH2016_genome_cost] for data on the cost of sequencing.*
+
+Many have sought to curb the costs of deep learning, with methods ranging from
+the very applied (e.g., reduced numerical precision [@tag:Gupta2015_prec
+@tag:Bengio015_prec @tag:Sa2015_buckwild @tag:Hubara2016_qnn]) to the exotic
+and theoretic (e.g., training small networks to mimic large networks and
+ensembles [@tag:Caruana2014_need @tag:Hinton2015_dark_knowledge]). The largest
+gains in efficiency have come from computation with graphics processing units
+(GPUs) [@tag:Raina2009_gpu @tag:Vanhoucke2011_cpu @tag:Seide2014_parallel
+@tag:Hadjas2015_cc @tag:Edwards2015_growing_pains
+@tag:Schmidhuber2014_dnn_overview], which excel at the matrix and vector
+operations so central to deep learning. The massively parallel nature of GPUs
+allows additional optimizations, such as accelerated mini-batch gradient
+descent [@tag:Vanhoucke2011_cpu @tag:Seide2014_parallel @tag:Su2015_gpu
+@tag:Li2014_minibatch]. However, GPUs also have a limited quantity of memory,
+which makes it difficult to train/use networks of significant size and
+complexity on a single GPU or machine [@tag:Raina2009_gpu
+@tag:Krizhevsky2013_nips_cnn]. This restriction has sometimes stymied the use
+of deep learning for computational biology[@tag:Chen2016_gene_expr] or
```

Comment:
Ahh! I see that now. In that case, I might suggest discussing it in just a little bit more detail. Something like:

> This restriction has sometimes forced computational biologists to use workarounds or to limit the size of an analysis. For example, Chen et al. [@tag:Chen2016_gene_expr] aimed to infer the expression levels of all genes, but due to memory restrictions they randomly partitioned genes into two halves and analyzed each separately. In other cases, researchers limited the size of their neural network [@tag:Wang2016_protein_contact]. Some have also chosen to use slower CPU implementations rather than sacrifice network size or performance [@tag:Yasushi2016_cgbvs_dnn].